lib/klaviyo.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var
+    Public = require('./public.js'),
+    Profiles = require('./profiles.js'),
+    Metrics = require('./metrics.js'),
+    Lists = require('./lists.js'),
+    {
+        KlaviyoError,
+        KlaviyoConfigurationError
+    } = require('./errors.js');
+
+const
+    WRITE_ACCESS_ERROR = 'This value is not writeable.';
+
+class Klaviyo {
+    constructor({ publicToken=null, privateToken=null } = {}) {
+        if (!publicToken && !privateToken) { 
+            throw new KlaviyoConfigurationError('You must provide a public or private api token.');
+        }
+
+        if (publicToken) {
+            this._public = new Public(publicToken);
+        }
+        if (privateToken) {
+            this._profiles = new Profiles(privateToken);
```

Comment:
bummer there isn't a way to do this more abstractly rather than setting the token for each thing?